File: php_eloquent/Entities/Course.php
<?php
declare(strict_types=1);

namespace App\Entities;

use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * Course
 *
 * @mixin Builder
 */
class Course extends Model
{
	protected $table = 'courses';
	public $timestamps = false;

	protected $fillable = ['name', 'description', 'available_from', 'available_to'];

	protected $hidden = ['pivot'];

	public function users() : BelongsToMany
	{
		return $this->belongsToMany(User::class, 'course_enrollments', 'course_id', 'user_id')
			->orderByDesc('account_role')
			->orderBy('surname');
	}

	public function tasks() : HasMany
	{
		return $this->hasMany(Task::class, 'course_id', 'id');
	}
}

File: dotnet/entity-framework_no-orm/Entities/Course.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;

namespace dotnet_entity_framework.Entities;

[Table("courses")]
public class Course
{
    [Key]
    public int Id { get; set; }
    public string Name { get; set; }
    public string? Description { get; set; }
    
    [JsonProperty(PropertyName = "available_from")]
    public DateTime AvailableFrom { get; set; }
    
    [JsonProperty(PropertyName = "available_to")]
    public DateTime? AvailableTo { get; set; }

    public List<User> Users { get; } = new();
    public ICollection<Task> Tasks { get; } = new List<Task>();
}
